packages/grafana-data/src/datetime/datemath.test.ts

```diff
@@ -183,6 +183,26 @@ describe('DateMath', () => {
     });
   });
 
+  describe('isMathString', () => {
+    it('should return true when valid date text', () => {
+      expect(dateMath.isMathString('now-1h')).toBe(true);
+    });
+
+    it('should return false when an absolute date is inserted', () => {
+      const date = new Date();
+      const result = dateMath.isMathString(date);
+      expect(result).toBe(false);
+    });
+
+    it('should return false when invalid date text', () => {
+      expect(dateMath.isMathString('2 + 2')).toBe(false);
+    });
+
+    it('should return false if no text is passed', () => {
+      expect(dateMath.isMathString('')).toBe(false);
```

Comment:
Does this account for whitespace too? If so, maybe add another assertion for a space` `

Reply:
Thank you for feedback , I have made the updates required based on the suggestions.